benchmarks/babelstream/src/babelStreamMainTest.cpp

```diff
@@ -267,210 +296,328 @@ void testKernels()
         bufAccInputBPtr,
         bufAccOutputCPtr);
 
-    // Vector of average run-times of babelstream kernels
-    std::vector<double> avgExecTimesOfKernels;
-    std::vector<double> minExecTimesOfKernels;
-    std::vector<double> maxExecTimesOfKernels;
-    std::vector<std::string> kernelLabels;
-    // Vector for collecting successive run-times of a single kernel in benchmark macro
-    std::vector<double> times;
+    auto const workDivNStream = alpaka::getValidWorkDiv(
+        kernelCfg,
+        devAcc,
+        NstreamKernel(),
+        bufAccInputAPtr,
+        bufAccInputBPtr,
+        bufAccOutputCPtr);
+
+    // To record runtime data generated while running the kernels
+    RuntimeResults runtimeResults;
 
     // Lambda for measuring run-time
     auto measureKernelExec = [&](auto&& kernelFunc, [[maybe_unused]] auto&& kernelLabel)
     {
-        for(auto i = 0; i < numberOfRuns; i++)
+        double runtime = 0.0;
+        auto start = std::chrono::high_resolution_clock::now();
+        kernelFunc();
+        alpaka::wait(queue);
+        auto end = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double> duration = end - start;
+        runtime = duration.count();
+        runtimeResults.kernelToRundataMap[kernelLabel]->timingsSuccessiveRuns.push_back(runtime);
+    };
+
+    // Run kernels one by one.
+    // Init kernel
+    alpaka::exec<Acc>(
+        queue,
+        workDivInit,
+        InitKernel(),
+        bufAccInputAPtr,
+        bufAccInputBPtr,
+        bufAccOutputCPtr,
+        static_cast<DataType>(initA),
+        static_cast<DataType>(initB));
```

Comment:
It is not one of babelstream kernels but ok i am implementing.

Reply:
Yes, I agree that it makes moderate sense at best but it might be interesting information and it brings us closer to upstream. Your announced change is not yet found in the PR.

Reply:
ok, implemented. Thanks.

```
AcceleratorType:AccGpuCudaRt<1,unsigned int>
NumberOfRuns:2
Precision:double
DataSize(items):1048576
DeviceName:NVIDIA RTX A500 Laptop GPU
WorkDivInit :{gridBlockExtent: (1024), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivCopy :{gridBlockExtent: (1024), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivMult :{gridBlockExtent: (1024), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivAdd  :{gridBlockExtent: (1024), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivTriad:{gridBlockExtent: (1024), blockThreadExtent: (1024), threadElemExtent: (1)}
WorkDivDot  :{gridBlockExtent: (256), blockThreadExtent: (1024), threadElemExtent: (1)}
Kernels         Bandwidths(GB/s) MinTime(s) MaxTime(s) AvgTime(s) DataUsage(MB) 
 AddKernel       87.201          0.0002886 0.0002886 0.0002886 25.166 
 CopyKernel      79.096          0.00021211 0.00021211 0.00021211 16.777 
 DotKernel       74.74           0.00022447 0.00022447 0.00022447 16.777 
 InitKernel      87.865          0.00028641 0.00028641 0.00028641 25.166 
 MultKernel      85.107          0.00019713 0.00019713 0.00019713 16.777 
 TriadKernel     87.046          0.00028911 0.00028911 0.00028911 25.166 
```